pkgs/stdenv/cross/default.nix

```
@@ -63,6 +63,8 @@ in lib.init bootStages ++ [
              # `tryEval` wouldn't catch, wrecking accessing previous stages
              # when there is a C compiler and everything should be fine.
              then throw "no C compiler provided for this platform"
+           else if crossSystem.isDarwin
```

How about?

```suggestion
           else if crossSystem.cc.isClang
```

(untested)

---

never mind. `cc` is set here in the first place.